packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -281,7 +289,7 @@ export class CdkToolkit {
           rollback: options.rollback,
           hotswap: options.hotswap,
           extraUserAgent: options.extraUserAgent,
-          buildAssets: assetBuildTime !== AssetBuildTime.ALL_BEFORE_DEPLOY,
+          buildAssets: assetBuildTime !== AssetBuildTime.ALL_BEFORE_DEPLOY, // TODO: remove this
```

Comment:
I flagged this before because I believe this no longer makes sense to have for `deployStack`. We probably want to hardcode it to `buildAssets: false` or remove it altogether.